### PR TITLE
cc38 complete

### PR DESCRIPTION
**Pull request by ryanapodaca**
- [x] Top-level README “Table of Contents” is updated
 - [x] README for this challenge is complete
       - [ ] Summary, Description, Approach & Efficiency, Solution
       - [ ] Picture of whiteboard
       - [ ] Link to code
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
       - [ ] “Happy Path” - Expected outcome
       - [ ] Expected failure
       - [ ] Edge Case (if applicable/obvious)

### CODE REVIEW

File: java/datastructures/lib/src/test/java/datastructures/graphs/TestGraph.java [hunk 1]
```diff
@@ -82,5 +91,43 @@ public class TestGraph {
...
+    expectedTraversalOrder.add(e);
+    expectedTraversalOrder.add(c);
+    expectedTraversalOrder.add(f);
+    expectedTraversalOrder.add(g);
+    expectedTraversalOrder.add(h);
+
+    LinkedList<Vertex<Character>> traversalOrder = graph.depthFirstTraversal(a);
+
+//    Assertions.assertEquals(expectedTraversalOrder, traversalOrder);
+    assertNull(graph.depthFirstTraversal(nullV));
```
Comment (lucy-gelderloos) [inline java/datastructures/lib/src/test/java/datastructures/graphs/TestGraph.java:130]: Your `depthFirstTraversal` method iterates over the list returned by `adjacencyLists.get(vertex)`. Because nullV is never added to a graph, it's never added to any `adjacencyLists`, so `adjacencyLists.get(nullV)` will return null. When Java tries to iterate over a null list, it crashes. You could fix it by adding a check to see if that list is null; from there you could handle it with an exception, by returning a null result, or something else.
